src/ParallelAlgorithms/LinearSolver/Multigrid/Tags.hpp

```diff
@@ -0,0 +1,205 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <optional>
+#include <vector>
+#include <unordered_set>
+
+#include "DataStructures/DataBox/PrefixHelpers.hpp"
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/Matrix.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/Tags.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "Options/Auto.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/PupStlCpp17.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace LinearSolver::multigrid {
+
+namespace OptionTags {
+
+template <typename OptionsGroup>
+struct MaxLevels {
+  using type = Options::Auto<size_t>;
+  static constexpr Options::String help =
+      "Maximum number of levels in the multigrid hierarchy. Includes the "
+      "finest grid, i.e. set to '1' to disable multigrids. Set to 'Auto' to "
```

Comment:
Does `lower_bound` work with `Options::Auto` to enforce the bound you have here?

Reply:
It doesn't :( But there's a validation check in `ElementsAllocator` that I had added for that reason.